### ISSUE TITLE
tv.mail.ru Siteconfig

### ISSUE THREAD

**Issue by UzunMuhalefet** [K-vanc/Tempest-EPG-Generator#98]
## Issue Type

Please choose your issue type

* [X] Request
* [ ] Bug
* [ ] Question

Hi, my friend.

I tried to create a site-config for tv.mail.ru to get Azerbaijani channels. Again I could not manage to solve JSON parsing :)
The deeper JSON elements are making my configuration wrong.

Let me go step by step.

**Channel Creation**
https://tv.mail.ru/ajax/channel/index/?page=##subpage1##&region_id=1096

- The region parameter refers to the geo-location and 1096 is for Baku.
- The subpage parameter starts from 0 and goes up to a total number that can be retrieved from the "total" JSON parameter.
- The channel bodies of the responses are not the same. Sometimes starts with the "has_live" parameter and sometimes starts with the "name" parameter directly. In addition, some sub-elements have the "name" parameter as well. So, I could not manage to get the actual channel name. For further steps, we need the ID value of each channel and the logo URL that can be retrieved from the parameters like "pic_url", "pic_url_128" or "pic_url_64".

**Channel Guide**
https://tv.mail.ru/ajax/channel/?channel_id=##channel##&date=##urldate1##

- The channel parameter is the ID of the channel retrieved from the previous response and the date parameter has a format like Y-m-d.
- The shows can be retrieved from the list of ["schedule"]["event"]["current"].
- Each show has the following useful attributes: "name", "start", "category_id", "id"
- The start field format is "H:i". I could not find category_id mapping. 
- Some shows are recommended and have ["recommended"]["image"]["src"] as a show icon and ["recommended"]["descr"] as a description.

**Show Detail**
Example URL: https://tv.mail.ru/ajax/event/?id=227559355&region_id=1096 

- I did not dive into this section. 
- The id parameter is the ID of the show retrieved from the previous response.

What I did is not working I am finding more channels and shows.

If you want to examine, I found configurations from different grabbers:
iptv-org grabber:
https://github.com/iptv-org/epg/blob/master/sites/tv.mail.ru/tv.mail.ru.config.js

WebGrab+ grabber:
https://github.com/hadi2291/myTV/blob/master/guide/siteini.pack/Russia/tv.mail.ru.ini

**Comment by K-vanc** [comment]
Hi,

I don't need any example from other places to make a siteconfig and just website name or adress is enough for reference.

But, i am not doing siteconfigs for Russian sources.

**Comment by UzunMuhalefet** [comment]
Sorry, I did not know that you have such a policy.

I am just preparing the Azerbaijan TV channels playlist. I am also working on EPG using your tool and creating new configs. Unfortunately, the best source for some Azerbaijani channels is tv.mail.ru. So, why I requested your help.

I shared other examples to make your job easy if you decide to work on this website. Nothing more than that.

Again sorry about that and thanks for your work and help.
